File: common/src/test/java/wile/redstonepen/client/GuisCoord2dTest.java
package wile.redstonepen.client;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;
import wile.redstonepen.client.Guis.Coord2d;

class GuisCoord2dTest
{
  @Test
  void constructorStoresXAndY()
  {
    final Coord2d c = new Coord2d(3, 7);
    assertEquals(3, c.x);
    assertEquals(7, c.y);
  }

  @Test
  void ofFactoryCreatesEquivalentCoord()
  {
    final Coord2d a = new Coord2d(5, 9);
    final Coord2d b = Coord2d.of(5, 9);
    assertEquals(a.x, b.x);
    assertEquals(a.y, b.y);
  }

  @Test
  void originIsZeroZero()
  {
    assertEquals(0, Coord2d.ORIGIN.x);
    assertEquals(0, Coord2d.ORIGIN.y);
  }

  @Test
  void toStringFormatsAsBracketedXCommaY()
  {
    assertEquals("[3,4]", new Coord2d(3, 4).toString());
  }

  @Test
  void toStringZeroZero()
  {
    assertEquals("[0,0]", Coord2d.ORIGIN.toString());
  }

  @Test
  void toStringNegativeValues()
  {
    assertEquals("[-1,-2]", new Coord2d(-1, -2).toString());
  }
}

File: common/src/test/kotlin/wile/redstonepen/client/GuisCoord2dTest.kt
package wile.redstonepen.client

import io.kotest.core.spec.style.DescribeSpec
import io.kotest.matchers.shouldBe
import wile.redstonepen.client.Guis.Coord2d

class GuisCoord2dTest :
    DescribeSpec({
        it("constructor stores x and y") {
            val c = Coord2d(3, 7)
            c.x shouldBe 3
            c.y shouldBe 7
        }

        it("of factory creates equivalent coord") {
            val a = Coord2d(5, 9)
            val b = Coord2d.of(5, 9)
            b.x shouldBe a.x
            b.y shouldBe a.y
        }

        it("ORIGIN is zero zero") {
            Coord2d.ORIGIN.x shouldBe 0
            Coord2d.ORIGIN.y shouldBe 0
        }

        it("toString formats as [x,y]") { Coord2d(3, 4).toString() shouldBe "[3,4]" }

        it("toString for zero zero is [0,0]") { Coord2d.ORIGIN.toString() shouldBe "[0,0]" }

        it("toString handles negative values") { Coord2d(-1, -2).toString() shouldBe "[-1,-2]" }
    })
